.clang-tidy

```diff
@@ -20,18 +20,23 @@ Checks: [
 
   '-bugprone-assignment-in-if-condition',
   '-bugprone-branch-clone',
+  '-bugprone-crtp-constructor-accessibility',
```

Comment:
Sorry, I commented already in the .clang-tidy file ... most new exclusions seem related to obscure edge cases. Could we try to fix them as much as possible in the code (similar to #79653)

Reply:
I'd wait first for perf test results to see if regression is gone, and if there are new ones.

New `clang-tidy` rules handling is at https://github.com/ClickHouse/ClickHouse/pull/79653.